app/models/manageiq/providers/lenovo/physical_infra_manager/event_catcher/stream.rb

```diff
@@ -27,12 +28,25 @@ def filter_fields
       { :operation => 'NOT', :field => 'eventClass', :value => '200' },
       { :operation => 'NOT', :field => 'eventClass', :value => '800' }
     ]
-    last_cn_event = get_last_cnn_from_events(@ems.id)
-    cn_operation = { :operation => 'GT', :field => 'cn', :value => last_cn_event.to_s }
-    fields.push(cn_operation) unless last_cn_event.nil?
+
+    unless @last_event_ems_ref.nil?
+      cn_operation = { :operation => 'GT', :field => 'cn', :value => @last_event_ems_ref.to_s }
+      fields.push(cn_operation)
+    end
     fields
   end
 
+  def parse_events(events)
+    if events.any?
```

Comment:
prefer `return if events.blank?` instead of having the whole method in a single conditional

Reply:
ok, done.